### PR TITLE
Add instruction for using dev image

### PR DESCRIPTION
**Pull request by chuanjin**
Signed-off-by: Chuan Jin <cjin@luxoft.com>

We move the instruction from downstream SWF since it is more generic and the gdb helper script is already upstreamed to pelux.

### CODE REVIEW

File: docs/articles/sde/how-to-use-dev-image.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
+:orphan:
+
+How to use development image
+===================================================
```
Comment (tobsan) [inline docs/articles/sde/how-to-use-dev-image.rst]: This underline is too long?

File: docs/articles/sde/how-to-use-dev-image.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+.. tags:: howto
+
+Introduction
+-------------
+The dev image is a superset of its corresponding production image for each variant, furthermore, it includes some more packages and tools, e.g. gdb, strace, etc. for the sake of developing an debugging convenience. Therefore, the dev image will be delivered together with the production image for each variant in the end.
```
Comment (tobsan) [inline docs/articles/sde/how-to-use-dev-image.rst]: This line is too long.

File: docs/articles/sde/how-to-use-dev-image.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+----------------------
+
+The handy QEMU launch script ``qemu_launcher`` is used to boot up dev image
+
+
+.. code-block:: bash
+
+   ./qemu_launcher -f FOLDER -i core-image-pelux-minimal-qemux86-64.ext4 --dev
+
+Note that the SSH port 22 is forwarded to 1234 by default, and an extra port 3333 is also exposed by the command for remote debugging later.
```
Comment (tobsan) [inline docs/articles/sde/how-to-use-dev-image.rst]: This line is also a bit long.

File: docs/articles/sde/how-to-use-dev-image.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    ./qemu_launcher -h
+
+The QEMU dev image can be accessed via SSH now
+
+.. code-block:: bash
+
+    $ ssh root@127.0.0.1 -p 1234
+
+
+The instructions above covered how to obtain the SDK and boot up QEMU dev image, steps below will show cross-compiling an example service (template-service) for the target, and then run a remote debugger on it.
```
Comment (tobsan) [inline docs/articles/sde/how-to-use-dev-image.rst]: This line is also a bit long.

File: docs/articles/sde/how-to-use-dev-image.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+Cross compile template-service
+-------------------------------
+
+Similar with image, SDK can be also fetched from Jenkins or local build.
```
Comment (tobsan) [inline docs/articles/sde/how-to-use-dev-image.rst]: Fix the grammar of this sentence.

File: docs/articles/sde/how-to-use-dev-image.rst [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+.. code-block:: bash
+
+    $ ssh root@127.0.0.1 -p 1234
+
+
+The instructions above covered how to obtain the SDK and boot up QEMU dev image, steps below will show cross-compiling an example service (template-service) for the target, and then run a remote debugger on it.
+
+Cross compile template-service
+-------------------------------
```
Comment (tobsan) [inline docs/articles/sde/how-to-use-dev-image.rst]: This underline is also a bit long.

File: docs/articles/sde/how-to-use-dev-image.rst [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+	[options] is any of the following:
+	  -f | --file            	Transfer the executable file to the target and start gdb
+	  -d | --dest            	The destination folder on target where the file will be transferred
+	  -P | --PID             	The PID on target to be attached
+	  -i | --target_ip       	The target IP address. Defaults to 127.0.0.1
+	  -p | --ssh_port        	SSH port to connect to QEMU. Defaults to 1234
+	  -t | --gdb_port        	The port for remote debugging. Defaults to 3333
+	  -s | --server          	Start gdbserver on target
+	  -g | --gdb             	Start gdb on development machine
+	  -h | --help            	Display this help
```
Comment (tobsan) [inline docs/articles/sde/how-to-use-dev-image.rst:120]: Do we really need this info here since it is available elsewhere, right?

**Review by tobsan — CHANGES_REQUESTED** [review]

**Comment by tobsan** [comment]
I also noticed that the CI complains about some words that need to be added to the wordlist.

File: docs/articles/sde/how-to-use-dev-image.rst (hunk 7 above)
Comment (chuanjin) [reply docs/articles/sde/how-to-use-dev-image.rst:120]: That's the other script for qemu launcher and this is the gdb helper

**Review by tobsan — APPROVED** [review]

**Comment by tobsan** [comment]
The CI reports an undefined label right now. Is there something that should be rebased or a bumped blueprint?

**Comment by chuanjin** [comment]
@tobsan  The label for "building-pelux-sources" is defined in SW, not in blueprint.

**Comment by tobsan** [comment]
@chuanjin well that explais it. This is the error message: `docs/articles/sde/how-to-use-dev-image.rst:20:undefined label: building-pelux-sources (if the link has no caption the label must precede a section header)`

You can't refer to a label in the SWF from the Blueprint.

**Pull request merged**